sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -4644,6 +4644,22 @@ object SQLConf {
       .booleanConf
       .createWithDefault(false)
 
+  val VARIANT_WRITE_SHREDDING_ENABLED =
+    buildConf("spark.sql.variant.writeShredding.enabled")
+      .internal()
+      .doc("When true, the Parquet reader is allowed to write shredded variant. ")
```

Comment:
Does this mean Delta is controlled some other way?

Reply:
The intent of this conf is to be a global kill switch for the write shredding feature, and that we'll pass a write-specific conf to actually control how we shred - e.g. have the shredding determined by sampling data in the task, or specify a shredding schema for each Variant column.

Right now, the conf only has an effect if `VARIANT_FORCE_SHREDDING_SCHEMA_FOR_TEST` is also set to a non-empty value. I could remove it if we don't think there's any value to having a global kill switch like this. @cloud-fan, any opinion?

Reply:
It's ok to have a kill switch.